Dockerfile

```diff
@@ -72,68 +71,131 @@ FROM builder-base AS pcov-builder
 FROM --platform=${BUILDPLATFORM} alpine AS toolbox-downloader
 WORKDIR /
 ARG TOOLBOX_VERSION
-RUN wget https://github.com/jakzal/toolbox/releases/download/v$TOOLBOX_VERSION/toolbox.phar
+RUN wget https://github.com/jakzal/toolbox/releases/download/v${TOOLBOX_VERSION}/toolbox.phar
 
 
 # PHP with all extensions needed, correctly configured
 FROM php-base as php-configured
 
-# Extensions .so
-COPY --link --from=ast-builder /usr/local/lib/php/extensions /usr/local/lib/php/extensions
-COPY --link --from=pcov-builder /usr/local/lib/php/extensions /usr/local/lib/php/extensions
-COPY --link --from=extensions-builder /usr/local/lib/php/extensions /usr/local/lib/php/extensions
-
-# Extensions .ini
-COPY --link --from=ast-builder /usr/local/etc/php/conf.d /usr/local/etc/php/conf.d
-COPY --link --from=pcov-builder /usr/local/etc/php/conf.d  /usr/local/etc/php/conf.d
-COPY --link --from=extensions-builder /usr/local/etc/php/conf.d /usr/local/etc/php/conf.d
+# Install extensions + extension config
+COPY --link --from=all-extensions /usr/local/lib/php/extensions /usr/local/lib/php/extensions
+COPY --link --from=all-extensions /usr/local/etc/php/conf.d /usr/local/etc/php/conf.d
 
+# Base php.ini
 COPY --link <<EOF /usr/local/etc/php/conf.d/phpqa.ini
 date.timezone=Europe/London
 memory_limit=-1
 phar.readonly=0
 pcov.enabled=0
 EOF
 
-COPY --link --chmod=755 <<EOF /entrypoint.sh
-#!/usr/bin/env sh
-set -e
+# Validate the PHP configuration
+RUN php --ini | grep phpqa.ini
 
-(test "\$GITHUB_ACTIONS" = "true" || test "\$CI" = "true") && test -f composer.json && composer install --no-scripts --no-progress
-
-exec "\$@"
-EOF
 ARG TOOLBOX_TARGET_DIR="/tools"
-COPY --link --from=toolbox-downloader --chmod=+x /toolbox.phar ${TOOLBOX_TARGET_DIR}/toolbox
-COPY --link --from=composer:2 /usr/bin/composer ${TOOLBOX_TARGET_DIR}/composer
+ENV TOOLBOX_TARGET_DIR="${TOOLBOX_TARGET_DIR}"
+WORKDIR ${TOOLBOX_TARGET_DIR}
+ENV PATH="${PATH}:${TOOLBOX_TARGET_DIR}"
+
+# Configure Composer
+COPY --link --from=composer:2 /usr/bin/composer ./composer
+ENV COMPOSER_HOME=${TOOLBOX_TARGET_DIR}/.composer
+ENV PATH="${PATH}:${COMPOSER_HOME}/vendor/bin"
+ENV COMPOSER_ALLOW_SUPERUSER 1
+RUN composer --version
 
-# INSTALLATION_DATE is used to bust the docker layer cache
-ARG INSTALLATION_DATE
+# Configure Toolbox
+COPY --link --from=toolbox-downloader --chmod=+x /toolbox.phar ./toolbox
 ARG TOOLBOX_EXCLUDED_TAGS
-ARG TOOLBOX_VERSION
 ENV TOOLBOX_EXCLUDED_TAGS=${TOOLBOX_EXCLUDED_TAGS}
-ENV TOOLBOX_TARGET_DIR="/tools"
+ARG TOOLBOX_VERSION
 ENV TOOLBOX_VERSION=${TOOLBOX_VERSION}
-ENV COMPOSER_ALLOW_SUPERUSER 1
-ENV COMPOSER_HOME=$TOOLBOX_TARGET_DIR/.composer
-ENV PATH="$PATH:$TOOLBOX_TARGET_DIR:${COMPOSER_HOME}/vendor/bin:$TOOLBOX_TARGET_DIR/QualityAnalyzer/bin:$TOOLBOX_TARGET_DIR/DesignPatternDetector/bin:$TOOLBOX_TARGET_DIR/EasyCodingStandard/bin"
+ENV PATH="${PATH}:${TOOLBOX_TARGET_DIR}/QualityAnalyzer/bin"
+ENV PATH="${PATH}:${TOOLBOX_TARGET_DIR}/DesignPatternDetector/bin"
```

Comment:
Can be removed, as the tool has been removed some time ago (https://github.com/jakzal/toolbox/pull/121).

Reply:
Yep, confirmed it can be removed: https://github.com/jakzal/toolbox/pull/440/files